### ISSUE TITLE
*Question* How Can I Create a GradientView From C#

### ISSUE THREAD

**Issue by mrobraven** [mgierlasinski/MagicGradients#145]
Hi,

I have looked at the readme and noticed the GradientBuilder part but have no idea how to take the result from that and turn it into an IGradientSource that the GradientView requires.

Basically, I have successfully designed a box in XAML but need to be able to generate the same box in c# but have no idea how to do this.

I know that there are a few issues that have been answered that are quite similar to this like #92, but they all relate to using the CSS side of your API which I am not, I just want a simple LinearGradient.

**My XAML**
`
<magicgradients:GradientView>
      <magicgradients:GradientView.GradientSource>
            <magicgradients:LinearGradient Angle="0">
                  <magicgradients:GradientStop Color="#ED68A0" Offset="0" />
                  <magicgradients:GradientStop Color="#E01446" Offset="1" />
            </magicgradients:LinearGradient>
      </magicgradients:GradientView.GradientSource>
</magicgradients:GradientView>
`

**C# So Far**
`
LinearGradientBuilder Gradients = new GradientBuilder().AddLinearGradient(0).AddStop(Color.FromHex("#ED68A0"))
.AddStop(Color.FromHex("#E01446"));
`

`GradientView gradientView = new GradientView{ };`

Basically this gradientView will then be added to a RelativeLayout which will be placed into a StackLayout.

**Comment by mgierlasinski** [comment]
Hi @mrobraven,
Documentation is being updated but it's still work in progress, please see readme on branch:
https://github.com/mgierlasinski/MagicGradients/tree/update-docs

To answer your question: there is an extension method `ToSource` that let's you set `IGradientSource` from `GradientBuilder`.

Please note that in next releases we plan to remove: `AddLinearGradient` and `AddRadialGradient` with parameters, it's a bit messy for Radial gradient. Please use overload with lambda:

``` c#
.AddLinearGradient(g => g
        .Rotate(20)
        .AddStops(Color.Red, Color.Green, Color.Blue))
```

**Comment by mrobraven** [comment]
Hi,

That's perfect thank you!

Thank you also for highlighting the changes included in the next release. I will keep a lookout for the next update and uncomment my new variant of the builder!

Many thanks and have a good Christmas!

Matthew
